src/stash/src/postgres.rs

```diff
@@ -61,24 +64,38 @@ CREATE TABLE uppers (
 /// tables are not specified and should not be relied upon. The only promise is
 /// stability. Any changes to the table schemas will be accompanied by a clear
 pub struct Postgres {
-    conn: Client,
+    url: String,
+    client: Option<Client>,
     epoch: i64,
+    rng: [u8; 16],
 }
 
 impl std::fmt::Debug for Postgres {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("Postgres")
+            .field("url", &self.url)
             .field("epoch", &self.epoch)
+            .field("rng", &self.rng)
             .finish_non_exhaustive()
     }
 }
 
 impl Postgres {
     /// Opens the stash stored at the specified path.
-    pub async fn open(mut conn: Client) -> Result<Postgres, StashError> {
-        conn.batch_execute("SET default_transaction_isolation = serializable")
-            .await?;
-        let tx = conn.transaction().await?;
+    pub async fn new(url: String) -> Result<Postgres, StashError> {
+        let mut conn = Postgres {
+            url,
+            client: None,
+            epoch: 0,
+            // The call to rand::random here assumes that the rng's seed source is from a
+            // secure source that will differ per thread. The docs for ThreadRng say it "is
+            // automatically seeded from OsRng", which meets this requirement.
+            rng: rand::random(),
```

Comment:
nit: this is a "nonce" or a "random number", right? "rng" makes me think "random number generator", not "randomly-generated number".